interactions/api/models/message.py

```diff
@@ -1026,11 +1026,16 @@ class Embed(DictSerializerMixin):
 
     def __init__(self, **kwargs):
         super().__init__(**kwargs)
-        self.timestamp = (
-            datetime.fromisoformat(self._json.get("timestamp"))
-            if self._json.get("timestamp")
-            else datetime.utcnow()
-        )
+        if self._json.get("timestamp"):
+            if isinstance(self._json.get("timestamp"), str):
+                self.timestamp = datetime.fromisoformat(
+                    self._json.get("timestamp")
+                )  # readability on non `_json` attr.
+            elif isinstance(self._json.get("timestamp"), datetime):
+                self._json["timestamp"] = str(
+                    self._json.get("timestamp")
+                )  # converts to isoformat for API.
+
```

Comment:
I made the `__setattr__` method to keep the `_json` the same as the attributes. Therefore, I don't like implementing a change between those two. Additionally, in my opinion, an isoformat can be but into the attribute, too. Converting to a datetime object is not that difficult for the user.